File: ferry_booking_app/lib/models/ticket.dart
class Ticket {
  final int id;
  final String ticketCode;
  final int bookingId;
  final int? passengerId;
  final int? vehicleId;
  final String qrCode;
  final String? seatNumber;
  final String boardingStatus;
  final String? boardingTime;
  final String status;
  final bool checkedIn;
  final String? watermarkData;
  final String? boardingGate;
  final String ticketType;
  final String createdAt;
  final String updatedAt;

  Ticket({
    required this.id,
    required this.ticketCode,
    required this.bookingId,
    this.passengerId,
    this.vehicleId,
    required this.qrCode,
    this.seatNumber,
    required this.boardingStatus,
    this.boardingTime,
    required this.status,
    required this.checkedIn,
    this.watermarkData,
    this.boardingGate,
    required this.ticketType,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Ticket.fromJson(Map<String, dynamic> json) {
    return Ticket(
      id: json['id'],
      ticketCode: json['ticket_code'],
      bookingId: json['booking_id'],
      passengerId: json['passenger_id'],
      vehicleId: json['vehicle_id'],
      qrCode: json['qr_code'],
      seatNumber: json['seat_number'],
      boardingStatus: json['boarding_status'],
      boardingTime: json['boarding_time'],
      status: json['status'],
      checkedIn: json['checked_in'] == 1 || json['checked_in'] == true,
      watermarkData: json['watermark_data'],
      boardingGate: json['boarding_gate'],
      ticketType: json['ticket_type'],
      createdAt: json['created_at'],
      updatedAt: json['updated_at'],
    );
  }
}

File: ferry_booking_app_New/app/Models/Ticket.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Ticket extends Model
{
    use HasFactory;

    protected $fillable = [
        'ticket_code',
        'booking_id',
        'passenger_id',
        'vehicle_id',
        'qr_code',
        'seat_number',
        'boarding_status',
        'boarding_time',
        'status',
        'checked_in',
        'watermark_data',
        'boarding_gate',
        'ticket_type',
    ];

    protected $casts = [
        'boarding_time' => 'datetime',
        'checked_in' => 'boolean',
    ];

    public function booking()
    {
        return $this->belongsTo(Booking::class);
    }

    public function passenger()
    {
        return $this->belongsTo(User::class, 'passenger_id');
    }

    public function vehicle()
    {
        return $this->belongsTo(Vehicle::class);
    }
}
